### PR TITLE
tests: add dcerpc test per #3109

### PR DESCRIPTION
**Pull request by catenacyber**
Rebase of #139 + adds `midstream` in configuration to get TCP rule working
https://redmine.openinfosecfoundation.org/issues/3109

### CODE REVIEW

File: tests/dcerpc/dcerpc-3109/test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# disables checksum verification
+args:
+  - -k none --set stream.midstream=true
+
+checks:
+
+  - filter:
+      count: 1
+      match:
+        event_type: dcerpc
```
Comment (inashivb) [inline tests/dcerpc/dcerpc-3109/test.yaml:11]: It's smb w dcerpc so the event would never be dcerpc, dcerpc will be nested inside smb event

Comment (catenacyber) [reply tests/dcerpc/dcerpc-3109/test.yaml:11]: Will you do the changes ? Or should I ? (and the suricata fixes if any)

Comment (inashivb) [reply tests/dcerpc/dcerpc-3109/test.yaml:11]: I will. Thank you for this :)
I'm chasing the bug as it still persists

Comment (catenacyber) [reply tests/dcerpc/dcerpc-3109/test.yaml:11]: What should I do with this PR ? Is there a redmine ticket reference ?

Comment (inashivb) [reply tests/dcerpc/dcerpc-3109/test.yaml:11]: I think it's I who needs to do things.. Will check the tickets.. Got a report on similar lines from Brandon and possibly Peter. Will review them and give the update/create the tickets. 
Sorry. this fell off the radar completely!

**Pull request closed without merging**